src/common/token.ts

```diff
@@ -140,23 +144,24 @@ export class ArweaveToken implements BaseToken<Transaction.default> {
               }
             }
           }
-        `,
-        })
-        .catch((err) => {
-          // Continue retries when request errors
-          this.logger.error('Failed to poll for transaction...', { err });
-          return undefined;
+          `,
         });
 
-      const transaction = response?.data?.data?.transaction;
+        transaction = response?.data?.data?.transaction;
+      } catch (err) {
+        // Continue retries when request errors
+        this.logger.debug('Failed to poll for transaction...', { err });
+        continue;
```

Comment:
with `continue` here - you do not wait to retry on errors - is that intended?

Reply:
good catch. not intended

Reply:
The circumstance only happens when I actually stub to throw though. typically a response is returned by arweave-js.api

Reply:
I think just removing the continue works 08a22f9